consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/validator/ForkingValidatorProviderTest.java

```diff
@@ -170,10 +173,50 @@ public void voteProviderIsDelegatesToHeadFork() {
         new ForkingValidatorProvider(
             blockChain, forksSchedule, blockValidatorProvider, contractValidatorProvider);
 
-    final VoteProvider voteProvider = Mockito.mock(VoteProvider.class);
-    when(contractValidatorProvider.getVoteProvider()).thenReturn(Optional.of(voteProvider));
+    validatorProvider.getVoteProviderAtHead();
 
-    assertThat(validatorProvider.getVoteProvider()).contains(voteProvider);
+    verify(contractValidatorProvider).getVoteProviderAtHead();
+    verifyNoInteractions(blockValidatorProvider);
+  }
+
+  @Test
+  public void voteProviderIsDelegatesToHeadFork_whenHeadIsBlockFork() {
+    final BftForksSchedule<QbftConfigOptions> forksSchedule =
+        new BftForksSchedule<>(createBlockFork(0), emptyList());
+
+    final ForkingValidatorProvider validatorProvider =
+        new ForkingValidatorProvider(
+            blockChain, forksSchedule, blockValidatorProvider, contractValidatorProvider);
+
+    validatorProvider.getVoteProviderAtHead();
```

Comment:
Same as above, doesn't look this is needed

Reply:
See other comment